rule/value.go

```diff
@@ -228,27 +229,46 @@ func ExprFromValue(val interface{}) bzl.Expr {
 		return be.BzlExpr()
 	}
 
-	rv := reflect.ValueOf(val)
-	switch rv.Kind() {
-	case reflect.Bool:
-		tok := "False"
-		if rv.Bool() {
-			tok = "True"
+	// Fast paths for common types to avoid reflection overhead.
+	switch v := val.(type) {
+	// common primitive types
+	case string:
+		return &bzl.StringExpr{Value: v}
+	case int:
+		return &bzl.LiteralExpr{Token: strconv.FormatInt(int64(v), 10)}
+	case int64:
+		return &bzl.LiteralExpr{Token: strconv.FormatInt(v, 10)}
+	case uint:
+		return &bzl.LiteralExpr{Token: strconv.FormatUint(uint64(v), 10)}
+	case uint64:
+		return &bzl.LiteralExpr{Token: strconv.FormatUint(v, 10)}
+	case float64:
+		return &bzl.LiteralExpr{Token: strconv.FormatFloat(v, 'g', -1, 64)}
+	case bool:
+		if v {
+			return &bzl.LiteralExpr{Token: "True"}
 		}
-		return &bzl.LiteralExpr{Token: tok}
+		return &bzl.LiteralExpr{Token: "False"}
+
+	// common types of slices
+	case []string:
+		return stringSliceToExpr(v)
+	case []interface{}:
+		return interfaceSliceToExpr(v)
+	}
 
-	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
-		reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+	// Fallback to reflection for less common types
+	rv := reflect.ValueOf(val)
+	switch rv.Kind() {
+	case reflect.Int8, reflect.Int16, reflect.Int32,
```

Comment:
All of these primitives could move really, I just did the ones that I (and magic AI robot) expected to be most common. WDYT?

Reply:
Move everything that can be moved I think. You might need a second type switch without the assignment (`switch val.(type)` rather than `switch v := val.(type)`) so the code can be shared by these cases. But it's better than reflection.